app/src/main/java/com/example/concordia_campus_guide/Fragments/PathInfoCardFragment.java

```diff
@@ -57,43 +46,44 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
 
         RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getContext());
         recyclerView.setLayoutManager(layoutManager);
-        directionList = new ArrayList<>();
+        updateDirectionResults();
+        return view;
+    }
 
+    private void updateDirectionResults() {
         Serializable temporaryDirectionsResults = getArguments().getSerializable("directionsResult");
-        directionsResults = (ArrayList<DirectionWrapper>) temporaryDirectionsResults;
-        walkingPointList = (ArrayList<WalkingPoint>) getArguments().getSerializable("walkingPoints");
-
-        pathInfoCardViewModel = new ViewModelProvider(this).get(PathInfoCardViewModel.class);
-
-
-        return view;
+        directionsResults = (ArrayList<PathInfoCard>) temporaryDirectionsResults;
     }
 
     @Override
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
-        getIndoorOutdoorInfo();
-        setStartAndEndTime();
+        totalDuration = 0;
+        //wait sufficient time for the google outdoor direction api callback
+        (new Handler()).postDelayed(this::drawIndoorOutdoorInfo, 2500);
```

Comment:
hmm is it the proper way of doing it? What I did to ensure I display everything in the Routes Overview section AFTER the api is done is by creating a directionsApiCallBack method in RoutesActivity that is called in the onPostExecute of DirectionsApiDataParser. Could you do something similar you think?

Reply:
We do have the callback in the PathsActivity. We had to add the timeout because the callback was coming late so the DirectionsWrapper list is being sent incomplete to the PathInfoCardFragment. If you know of any better way to do it please let us know because we had so much trouble with this.

Reply:
hmm i see, then it's fine like that then